scaleway/resource_instance_server_test.go

```diff
@@ -881,3 +882,133 @@ func TestAccScalewayInstanceServer_ServerWithBlockNonDefaultZone(t *testing.T) {
 		},
 	})
 }
+
+func TestAccScalewayInstanceServer_PrivateNetwork(t *testing.T) {
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckScalewayInstanceServerDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource scaleway_vpc_private_network pn01 {
+						name = "private_network_instance"
+					}
+			
+					resource "scaleway_instance_server" "base" {
+					  image = "ubuntu_focal"
+					  type  = "DEV1-S"
+			
+				  	  tags = [ "private_network" ]
+					  private_network {
+						pn_id = scaleway_vpc_private_network.pn01.id
+					  }
+					}`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayInstanceServerExists(tt, "scaleway_instance_server.base"),
```

Comment:
Could you also check that the change happened on scaleway side too? With a `testAccCheckScalewayInstancePrivateNetworkAttachmentExists`